code/clue.js

```diff
@@ -1,59 +1,374 @@
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  if (!killerIsPicked) {
+    document.getElementById('loaderKiller').style.opacity = 1
+    setTimeout(function () {
+      document.getElementById('loaderKiller').style.opacity = 0
+      shuffleFavoriteWeapon();
+      mystery.killer = randomSelector(suspects)
+      document.getElementById('killerName').style.background = mystery.killer.color
+      document.getElementById(
+        'killerName'
+      ).innerHTML = `Name: ${mystery.killer.firstName} ${mystery.killer.lastName}`
+      document.getElementById('killerAge').innerHTML = (`Age: ${mystery.killer.age}`)
+      document.getElementById('killerOccupation').innerHTML = `Occupation: ${mystery.killer.occupation}`
+      document.getElementById('killerDescription').innerHTML = `Descripton: ${mystery.killer.description}`
+      document.getElementById('killerFavouriteWeapon').innerHTML = `Preferred weapon: ${mystery.killer.favouriteWeapon}` //red level to check that the fav-weap changes by using function "shuffleFavouriteWeapon"
+      document.getElementById('killerImage').src = mystery.killer.image;
+      document.getElementById('killerImage').style.opacity = 1;
+      killerIsPicked = true;
+    }, 2500)
+  }
+};
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+// This function will be invoked when you click the weapon card.
+const pickWeapon = () => {
+  if (killerIsPicked && !weaponIsPicked) {
+    document.getElementById('loaderWeapon').style.opacity = 1
+    setTimeout(function() {
+      document.getElementById('loaderWeapon').style.opacity = 0
+      let chanceFavWeapon = chanceOfFavouriteWeapon();
+      mystery.weapon = randomSelector(chanceFavWeapon)
+      document.getElementById('weaponName').innerHTML = `Type: ${mystery.weapon.name}`
+      document.getElementById('weaponWeight').innerHTML = `Weight: ${mystery.weapon.weight}`
+      weaponIsPicked = true
+    }, 2500)
+  }else alert('Found out who the killer is first');
+};
+
+//This function will be invoked when you click  the room card.
+const pickRoom = () => {
+  if (weaponIsPicked && !roomIsPicked) {
+    document.getElementById('loaderRoom').style.opacity = 1
+    setTimeout(function() { 
+      document.getElementById('loaderRoom').style.opacity = 0
+      let chanceForSuspectedRoom = chanceOfRoom();
+      mystery.room = randomSelector(chanceForSuspectedRoom) 
+      document.getElementById('roomName').innerHTML = `Where murder was commited: ${mystery.room.name}`
+      roomsIsPicked = true
```

Comment:
Semantically, it should rather be `isRoomPicked` than `roomIsPicked`

Reply:
Correct, it makes a lot more sense. I have changed that now.